compat/operator/reduce.ts

```diff
@@ -2,9 +2,8 @@ import { Observable } from 'rxjs';
 import { reduce as higherOrderReduce } from 'rxjs/operators';
 
 /* tslint:disable:max-line-length */
+export function reduce<T, R>(this: Observable<T>, accumulator: (acc: R, value: T, index: number) => R, seed?: R): Observable<R>;
```

Comment:
I don't think the seed can be optional if the accumulator has a different type the the source elements. If a seed is not specified, my understanding is that the first element will be used, instead. That means that the initial accumulator value will be a `T` and not an `R` as indicator by the call signature.

Reply:
You're correct... adding tests and updating.